extensions/big-data-cluster/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -0,0 +1,500 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import { ClusterController, IEndPoint, IControllerError } from '../controller/clusterController';
+import { ControllerTreeDataProvider } from '../tree/controllerTreeDataProvider';
+import { TreeNode } from '../tree/treeNode';
+
+const localize = nls.loadMessageBundle();
+
+export class AddControllerDialogModel {
+	constructor(
+		public treeDataProvider: ControllerTreeDataProvider,
+		public node?: TreeNode,
+		public prefilledUrl?: string,
+		public prefilledUsername?: string,
+		public prefilledPassword?: string,
+		public prefilledRememberPassword?: boolean
+	) {
+		this.prefilledUrl = prefilledUrl || (node && node['url']);
+		this.prefilledUsername = prefilledUsername || (node && node['username']);
+		this.prefilledPassword = prefilledPassword || (node && node['password']);
+		this.prefilledRememberPassword = prefilledRememberPassword || (node && node['rememberPassword']);
+	}
+
+	public async onComplete(url: string, username: string, password: string, rememberPassword: boolean): Promise<void> {
+		let clusterController = new ClusterController();
+		let response = await clusterController.getEndPoints(url, username, password, true);
```

Comment:
This code makes me think the ClusterController should contain fields for url, username, password and be the thing that `addController` accepts? Just a thought- your implementation is fine for now

Reply:
ClusterController is not a controller. It is just a namespace that contains client API methods.

Reply:
I will remove the class since the name is confusing.